modules/consensus/consensusset.go

```diff
@@ -176,7 +177,10 @@ func (cs *ConsensusSet) ChildTarget(id types.BlockID) (target types.Target, exis
 func (cs *ConsensusSet) Close() error {
 	cs.mu.Lock()
 	defer cs.mu.Unlock()
-	return cs.db.Close()
+	return build.JoinErrors([]error{
+		build.AddContext("log.Close failed:", cs.log.Close()),
+		build.AddContext("db.Close failed:", cs.db.Close()),
+	}, "; ")
```

Comment:
This is interesting, I'm not sure I like the construction. I like both the context and the fact that we're joining the errors instead of returning early, but not necessarily the actual way the code is constructed. You are returning a function that's taking an array of function calls that have function calls as arguments.

Doesn't seem like the right way to do things.

Also, if you put the last argument of `JoinErrors` as the first argument, you could make the array variadic, which I think is more appropriate for `JoinErrors` anyway.